py/server.py

```diff
@@ -365,7 +367,7 @@ def update(p, args):
         idxs = list(range(len(pdata)))
 
         if name is not None:
-            assert len(new_data['x']) == 1
+            assert len(new_data) == 1
```

Comment:
Does this change break the existing ability to use updateTrace? If so, can you use a case to maintain the old functionality of updateTrace and stick a `#TODO remove when updateTrace is deprecated` there (can be done by just copying the old function to a new def, putting a case here, and calling the old one for the old case)?